pkg/controllermanager/controller/event/event_control.go

```
@@ -0,0 +1,76 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package event
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/gardener/gardener/pkg/client/kubernetes/clientmap/keys"
+	"github.com/gardener/gardener/pkg/logger"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/tools/cache"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var nowFunc = time.Now
+
+func (c *Controller) enqueueEvent(obj interface{}) {
+	key, err := cache.MetaNamespaceKeyFunc(obj)
+	if err != nil {
+		logger.Logger.Errorf("Couldn't get key for object %+v: %v", obj, err)
+		return
+	}
+	c.eventQueue.Add(key)
+}
+
+func (c *Controller) reconcileEvent(req reconcile.Request) (reconcile.Result, error) {
+	ctx := context.TODO()
+
+	gardenClient, err := c.clientMap.GetClient(ctx, keys.ForGarden())
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to get garden client: %w", err)
+	}
+
+	log := logger.NewFieldLogger(logger.Logger, "event", req.Name)
+
+	event := &corev1.Event{}
+	if err := gardenClient.Client().Get(ctx, req.NamespacedName, event); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Debug("Skipping because Event has been deleted")
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, err
+	}
+
+	if event.InvolvedObject.Kind == "Shoot" {
+		return reconcile.Result{}, nil
+	}
+
+	deleteAt := event.FirstTimestamp.Add(c.cfg.Controllers.Event.TTLNonShootEvents.Duration)
+	timeUntilDeletion := deleteAt.Sub(nowFunc())
+	if timeUntilDeletion > 0 {
+		return reconcile.Result{RequeueAfter: timeUntilDeletion}, nil
+	}
+
+	err = gardenClient.Client().Delete(ctx, event)
+	if err != nil {
+		log.Debug("Failed to delete event due to %w", err)
```

Should we rather return the error here? It will then be logged only once in all cases, now it is logged twice in case  log level is set to `debug`.